logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/call/usecase/ObserveEndCallDialogUseCase.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.call.usecase
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+interface ObserveEndCallDialogUseCase {
+    suspend operator fun invoke(): Flow<Unit>
+}
+
+internal class ObserveEndCallDialogUseCaseImpl(
+    private val dialogManager: EndOngoingCallManager
+) : ObserveEndCallDialogUseCase {
+    override suspend fun invoke(): Flow<Unit> =
+        dialogManager.observeCallEndedBecauseOfVerificationDegraded().map { Unit }
+}
```

Comment:
Do we need to create a new class, can't we just use `EndOngoingCallManager` directly in other usecases ?
Also for naming, can we remove the term `dialog` as Kalium is purly logic and there is no UI

Reply:
this feature is not well documented yet, maybe later we'll need to add some info about the conversation (name, users that caused verification degrade, etc.)
In that case `EndCallResultListener` is only for listening and observing events, and `ObserveEndCallDialogUseCaseImpl` is for collecting that data and propagating it to the platform side.